lib/src/meta.dart

```diff
@@ -25,104 +31,109 @@ typedef GeomEachCallback = dynamic Function(
-    geometryMaybeCollection = (isFeatureCollection
-        ? geoJSON.features[i].geometry
-        : (isFeature ? geoJSON.geometry : geoJSON));
-    featureProperties = (isFeatureCollection
-        ? geoJSON.features[i].properties
-        : (isFeature ? geoJSON.properties : {}));
-    featureBBox = (isFeatureCollection
-        ? geoJSON.features[i].bbox
-        : (isFeature ? geoJSON.bbox : null));
-    featureId = (isFeatureCollection
-        ? geoJSON.features[i].id
-        : (isFeature ? geoJSON.id : null));
-    isGeometryCollection = (geometryMaybeCollection != null)
-        ? geometryMaybeCollection.type == GeoJSONObjectTypes.geometryCollection
-        : false;
-    stopG =
-        isGeometryCollection ? geometryMaybeCollection.geometries.length : 1;
+void geomEach(GeoJSONObject geoJSON, GeomEachCallback callback) {
+  try {
+    if (geoJSON is FeatureCollection) {
+      _forEachGeomInFeatureCollection(geoJSON, callback);
+    } else if (geoJSON is Feature) {
+      _forEachGeomInFeature(geoJSON, callback, 0);
+    } else if (geoJSON is GeometryObject) {
+      _forEachGeomInGeometryObject(geoJSON, callback, {}, null, null, 0);
+    } else {
+      throw Exception('Unknown Geometry Type');
+    }
+  } on _ShortCircuit {
+    return;
+  }
+}
+
+void _forEachGeomInFeatureCollection(
+    FeatureCollection featureCollection, GeomEachCallback callback) {
+  int featuresLength = featureCollection.features.length;
+  for (int featureIndex = 0; featureIndex < featuresLength; featureIndex++) {
+    _forEachGeomInFeature(
+        featureCollection.features[featureIndex], callback, featureIndex);
+  }
+}
+
+void _forEachGeomInFeature(Feature<GeometryObject> feature,
+    GeomEachCallback callback, int featureIndex) {
+  _forEachGeomInGeometryObject(feature.geometry!, callback, feature.properties!,
+      feature.bbox, feature.id, featureIndex);
+}
 
-    for (var g = 0; g < stopG; g++) {
-      geometry = isGeometryCollection
-          ? geometryMaybeCollection.geometries[g]
-          : geometryMaybeCollection;
-      if (geometry == null) {
+void _forEachGeomInGeometryObject(
+    GeometryObject geometryObject,
+    GeomEachCallback callback,
+    Map<String, dynamic> featureProperties,
+    BBox? featureBBox,
+    dynamic featureId,
+    int featureIndex) {
+  GeometryType currentGeometry;
+  num geometryCollectionLength = _getGeometryCollectionLength(geometryObject);
+  for (int geometryIndex = 0;
+      geometryIndex < geometryCollectionLength;
+      geometryIndex++) {
+    currentGeometry = _getGeometry(geometryObject, geometryIndex);
+    _runGeomEachCallbacks(currentGeometry, callback, featureIndex,
+        featureProperties, featureBBox, featureId);
+  }
+}
+
+void _runGeomEachCallbacks(
+    GeometryType<dynamic> currentGeometry,
+    GeomEachCallback callback,
+    int featureIndex,
+    Map<String, dynamic> featureProperties,
+    BBox? featureBBox,
+    dynamic featureId) {
+  switch (currentGeometry.type) {
```

Comment:
In my PR #33, I used the `is` operator instead of checking the `.type` attribute in a switch case, which could save us some lines of code. Don't know if type checking is more efficient than this.

like this:
```dart
if(currentGeometry is GeometryType) {
// ...
} else if(currentGeometry is GeometryCollection) {
// ...
} else {
    throw Exception('Unknown Geometry Type');
}
```